### PR TITLE
Adding report processing to SBFGame+Manager, add Report interfaces

### PR DESCRIPTION
**Pull request by SJuliez**
This is a further step in the SBF direction. It
- modernizes Report processing a bit by replacing Vector with List
- adds interfaces for the full report and the individual entries (these are not terribly important as report types aren't necessarily interchangeable but they do help focus on which methods are important and which are implementation dependent)
- adds a full report subclass for SBF (the main difference to the old GameReport is that it uses a map of game round->content instead of just a list of content)
- adds a bit of report processing to the SBF Game and Manager

I ran the tests and had Princess play a full game which stalled as Princess was unwilling to finish off a retreating tank that was immobile. Doesn't seem like a problem of this PR :)

### CODE REVIEW

**Comment by Sleet01** [comment]
Is this superseded by #5466?

**Comment by SJuliez** [comment]
> Is this superseded by #5466?

No (you probably guessed it already from my answers over there), the other one just builds on this one. Sorry for the confusion.

File: megamek/src/megamek/common/SBFFullGameReport.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        fullReport.clear();
+    }
+
+    public void replaceAllReports(Map<Integer, List<Report>> newReports) {
+        clear();
+        fullReport.putAll(newReports);
+    }
+
+
+    public Map<Integer, List<Report>> createFilteredReport(Player recipient) {
```
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/common/SBFFullGameReport.java:58]: This is very preliminary code so far but obviously the recipient will be used for filtering the report at some point.

**Review by Sleet01 — APPROVED** [review]
Looks good.  I think any questions are covered in the subsequent PR.

**Pull request merged**